controllers/appmesh/gatewayroute_controller.go

```diff
@@ -83,7 +89,11 @@ func (r *gatewayRouteReconciler) reconcile(ctx context.Context, req ctrl.Request
 	if !gr.DeletionTimestamp.IsZero() {
 		return r.cleanupGatewayRoute(ctx, gr)
 	}
-	return r.reconcileGatewayRoute(ctx, gr)
+	if err := r.reconcileGatewayRoute(ctx, gr); err != nil {
+		r.recorder.Event(gr, corev1.EventTypeWarning, "ReconcileError", err.Error())
```

Comment:
I'm not convinced that this should be `WARN` level, failing to create any resource will technically impact the routing in the cluster. Thoughts @jeremymill and @cgchinmay 

Also, could we rename `ReconcileError`, can we get the error type from the `err` Object? Could it be specific such as`BadRequestException` or `LimitExceededException` or whatever else?

Reply:
Since we only have Warning events and Normal events, I'm not sure if Normal is more appropriate than Warning here.

I could make a utility function to extract the error type (`err.split(":")[0]`, effectively) but this probably won't behave well unless all errors are formatted this way.

Reply:
@jeremymill EventType is a string so you can pass "Error" string instead or have our own custom ErrorType to distinguish between Errors and warnings. Will be helpful in future to decide based on the error type.

Reply:
I tried passing "Error" string instead of EventType, this causes the event to not appear.

Reply:
Okay, will have to check where is this validated since it was a string type. But in that case, I guess we will have to use only the built-in EventTypes for now

Reply:
That's lame that `EventType` doesn't support other types, let's keep it `WARN` for now and come back and fix it.

What members/methods does the `err` Object have? I understand that `err.Error()` is a string, but does it also have `err.Type()`? If not, let's leave it at `ReconcileError`, substrings are hacky.